### PR TITLE
EXP-3026 EXP-3029 fix device getLocation and getExperience make api call

### PR DESCRIPTION
**Pull request by cesar-oyarzun-m**

### CODE REVIEW

File: exp-android-sdk/src/main/java/com/scala/exp/android/sdk/model/Device.java [hunk 1]
```diff
@@ -25,17 +25,29 @@ public void setZones(List<Zone> zones) {
...
     }
 
-    public Location getLocation() {
-        return location;
+    public ExpObservable<Location> getLocation() {
+        ExpObservable<Location> expObs= null;
+        if(this.location!=null && this.location.getUuid()!=null){
+            expObs =Exp.getLocation(this.location.getUuid());
+        }else{
+            expObs = new ExpObservable<Location>(Observable.<Location>empty());
```
Comment (adamgallowayscala) [inline exp-android-sdk/src/main/java/com/scala/exp/android/sdk/model/Device.java]: 👍

File: exp-android-sdk/src/main/java/com/scala/exp/android/sdk/model/Experience.java [hunk 2]
```diff
@@ -26,13 +26,13 @@
 
     public ExpObservable<Experience> getCurrentExperience(){
         final ExpObservable<Device> observable = Device.getCurrentDevice();
-            return new ExpObservable<Experience>(observable.<Experience>flatMap(new Func1<Device, Observable<Experience>>() {
+            return new ExpObservable<Experience>(observable.<Experience>flatMapExp(new Func1<Device, ExpObservable<Experience>>() {
                 @Override
-                public Observable<Experience> call(Device device) {
+                public ExpObservable<Experience> call(Device device) {
                     if(device!=null && device.getExperience()!=null){
```
Comment (adamgallowayscala) [inline exp-android-sdk/src/main/java/com/scala/exp/android/sdk/model/Experience.java]: won't calling getExperience() for the null check call the api twice?  this should probably check the document like device.get('experience.uuid') or something similar right?

**Comment by cesar-oyarzun-m** [comment]
ok I think we're good now 👍

**Comment by adamgallowayscala** [comment]
👍

**Pull request merged**